packages/components/src/test/properties/NavigationPropertyTargetSelector.test.tsx

```diff
@@ -168,16 +169,124 @@ describe("NavigationPropertyTargetSelector", () => {
       getNavigationPropertyInfo: async () => testNavigationPropertyInfo,
       propertyRecord,
     };
-    const { rerender, queryByText } = render(<NavigationPropertyTargetSelector {...initialProps} />);
+    const { rerender, queryByDisplayValue } = render(<NavigationPropertyTargetSelector {...initialProps} />);
 
-    await waitFor(() => expect(queryByText(value.displayValue)).to.not.be.null);
+    await waitFor(() => expect(queryByDisplayValue(value.displayValue)).to.not.be.null);
     const newValue = {
       valueFormat: PropertyValueFormat.Primitive,
       value: { id: "2", className: "TestSchema:TargetClass" },
       displayValue: "New Target Class Instance",
     };
     const newPropertyRecord = new PropertyRecord(newValue as PropertyValue, propertyDescription);
     rerender(<NavigationPropertyTargetSelector {...initialProps} propertyRecord={newPropertyRecord} />);
-    await waitFor(() => expect(queryByText(newValue.displayValue)).to.not.be.null);
+    await waitFor(() => expect(queryByDisplayValue(newValue.displayValue)).to.not.be.null);
+  });
+
+  it("click on dropdown button closes and opens menu", async () => {
+    sinon.stub(Presentation.presentation, "getContent").resolves(new Content(createTestContentDescriptor({ fields: [], categories: [] }), []));
+    const propertyDescription = createNavigationPropertyDescription();
+    const value = {
+      valueFormat: PropertyValueFormat.Primitive,
+      value: { id: "1", className: "TestSchema:TargetClass" },
+      displayValue: "Target Class Instance",
+    };
+    const propertyRecord = new PropertyRecord(value as PropertyValue, propertyDescription);
+
+    const initialProps: NavigationPropertyTargetSelectorProps = {
+      imodel: testImodel,
+      getNavigationPropertyInfo: async () => testNavigationPropertyInfo,
+      propertyRecord,
+    };
+    const { container } = render(<NavigationPropertyTargetSelector {...initialProps} />);
+
+    const dropdownButton = await waitFor(() => {
+      const element = container.querySelector<HTMLDivElement>(".iui-end-icon");
+      expect(element).to.not.be.null;
+      return element;
+    });
+
+    fireEvent.click(dropdownButton!);
+
+    await waitFor(() => expect(container.querySelector<HTMLDivElement>(".iui-menu")).to.not.be.null);
```

Comment:
I think it would be better to check if menu items itself are visible instead. Something like `queryByText(itemLabel)`

Same in other tests where checking if menu is open or not.

Reply:
fixed